tldw_Server_API/app/core/Setup/install_manager.py

```
@@ -442,8 +474,382 @@ def execute_install_plan(plan_payload: dict[str, Any]) -> None:
+        catalog_version=bundle.catalog_version,
+    )
+    completed_steps = _completed_step_names(get_install_status_snapshot()) if safe_rerun else set()
+    if safe_rerun and expected_steps and expected_steps.issubset(completed_steps):
+        for step_name in sorted(expected_steps):
+            result_steps.append(
+                {
+                    "name": step_name,
+                    "status": "skipped",
+                    "detail": "Already satisfied by a previous successful install run.",
+                }
+            )
```

Ruff (0.15.6)

[warning] Use `list.extend` to create a transformed list

(PERF401)